File: Python/Recursions/squarest_game_on_beach.py
def solve(goal, standing, score=0, dropped=[]):
    if goal == score:
        return dropped
    elif score > goal or len(standing) == 0:
        return None
    else:
        chosen = standing[0]
        others = standing[1:]
        return solve(goal, others, score + chosen, dropped + [chosen]) or solve(
            goal, others, score, dropped
        )


print(solve(50, [15, 9, 30, 21, 19, 3, 12, 6, 25, 27]))

File: Go/Recursions/squarestGameOnBeach.go
package main

import (
	"fmt"
)

func solve(goal int, standing []int, score int, dropped []int) []int {
	if score == goal {
		return dropped
	} else if score > goal || len(standing) == 0 {
		return nil
	} else {
		chosen := standing[0]
		others := standing[1:]

		if result := solve(goal, others, chosen+score, append(append([]int{}, dropped...), chosen)); result != nil {
			return result
		}
		return solve(goal, others, score, dropped)
	}

}

func main() {
	result := solve(50, []int{15, 9, 30, 21, 19, 3, 12, 6, 25, 27}, 0, []int{})
	fmt.Println(result)
}